grails-app/controllers/au/org/emii/aatams/SensorController.groovy

```diff
@@ -32,8 +32,6 @@ class SensorController extends ReportController {
         def tag = tagFactoryService.lookupOrCreate(params.tag)
         assert(tag)
 
-        params.tag = tag
```

Comment:
The sensor (Child) object should be assigned to Tag (Parent) object using the following code.
```
tag.addToSensors(sensorInstance)  (Updates both parent and child objects)
```

Therefore, I think we do not need to update the (Tag) parameters during the creation of the sensor object as this was confusing hibernate. I am not sure how/why it was working earlier. 
```
params.tag = tag  (Not Needed)
```

Did you get a chance to try out the code on your local environment?